app/frontend/components/ConcertoRichText.vue

```diff
@@ -25,38 +25,48 @@ const container = ref();
 const child = ref();
 
 async function resizeText() {
-  let containerElement = container.value;
-  let contentElement = child.value;
+  const containerElement = container.value;
+  const contentElement = child.value;
 
-  let displayHeight = contentElement.scrollHeight;
-  let fieldHeight = containerElement.offsetHeight;
-  let fontScale = 100;
+  const fieldHeight = containerElement.offsetHeight;
+  const initialHeight = contentElement.scrollHeight;
 
-  if (displayHeight === 0 || fieldHeight === 0) {
+  if (initialHeight === 0 || fieldHeight === 0) {
     // Nothing to do.
     console.error('Cannot resize text: zero height detected.');
     return;
   }
 
-  console.debug(`Field height: ${fieldHeight}, Display height: ${displayHeight}`);
-
-  // First expand the content to fill the height of the div.
-  while (displayHeight < fieldHeight) {
-    container.value.style.fontSize = `${fontScale}%`;
-    fontScale++;
-
-    displayHeight = contentElement.scrollHeight;
+  console.debug(`Field height: ${fieldHeight}, Initial content height: ${initialHeight}`);
+
+  // Use binary search to find optimal font size
+  // This reduces layout reflows from O(n) to O(log n)
+  let minSize = 1;
+  let maxSize = 500; // Maximum 500% font size
```

Comment:
This sets a new max font size.  In my tests, this is not large enough for short strings "Hello World" in large content fields like 1000px x 1000px - remember, this plays on TV-sized screens.  We probably need to make this number bigger?